src/main/scala/org/apache/spark/OrderedPartitioner.scala

```diff
@@ -0,0 +1,89 @@
+package org.apache.spark
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+
+import org.apache.spark.util.CollectionsUtils
+import org.broadinstitute.hail.variant.Variant
+
+import scala.reflect.ClassTag
+
+/**
+  * A [[org.apache.spark.Partitioner]] that partitions sortable records by range into roughly
+  * equal ranges. The ranges are determined by sampling the content of the RDD passed in.
+  *
+  * Note that the actual number of partitions created by the OrderedPartitioner might not be the same
+  * as the `partitions` parameter, in the case where the number of sampled records is less than
+  * the value of `partitions`.
+  */
+case class OrderedPartitioner[T: Ordering : ClassTag, K: Ordering : ClassTag](
+  rangeBounds: Array[T],
+  private var ascending: Boolean = true)(implicit f: (K) => T)
```

Comment:
We have mostly been using the extra parens.  I have, at least.  I'll add it to the style guide.